src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2011,6 +2011,7 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
             {
                 // Delegate targets are always treated as direct calls here. (It would be nice to clean it up...).
                 directCall = true;
+                resolvedCallVirt = !targetMethod.IsVirtual || targetMethod.IsFinal || targetMethod.OwningType.IsSealed();
```

Comment:
I would ideally want to fix this in the exact same way CoreCLR handles this, but I can't find where that happens (they clearly don't set the bool on CoreCLR even though they check for it and expect it to represent the reality).

I think there's value in trying to mirror the CoreCLR implementation and don't diverge too much because this method will get more complicated over time and the CoreCLR JitInterface has a decade of bug fixes in it (and admittedly, a decade of legacy quirks).

I will go with this as a minimally invasive fix to the structure of the method. We can always revisit.